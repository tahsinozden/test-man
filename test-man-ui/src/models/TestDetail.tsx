class TestDetail {
    constructor(public id?: number, public name?: string, public status?: string, public description?: string) {
    }
}

export default TestDetail;