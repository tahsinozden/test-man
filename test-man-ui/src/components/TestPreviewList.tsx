import React, {useEffect, useState} from "react";
import TestDetail from "../models/TestDetail";
import {Container} from "react-bootstrap";
import TestPreview from "./TestPreview";

interface TestPreviewListProps {
    testDetails: TestDetail[];
    onTestDataChange: () => void;
}

const TestPreviewList: React.FC<TestPreviewListProps> = ({testDetails, onTestDataChange}: TestPreviewListProps) => {
    const [details, setDetails] = useState<any>([]);
    let items: any[] = [];

    useEffect(() => {
        for (let i in testDetails) {
            items.push(
                <TestPreview key={i} testDetail={testDetails[i]} onStatusSelect={updateTestStatus}></TestPreview>
            )
            setDetails(items);
        }
    }, [testDetails]);

    const updateTestStatus = (testDetail: TestDetail, newStatus: string) => {
        const newTestDetail = {...testDetail};
        newTestDetail.status = newStatus;
        submitStatus(testDetail);
    }

    const submitStatus = (testDetail: TestDetail) => {
        const requestOptions = {
            method: 'PATCH',
            headers: {'Content-Type': 'application/json'},
            body: JSON.stringify({status: testDetail.status})
        };
        // TODO: URL from configs
        fetch(`http://localhost:8080/api/v1/tests/${testDetail.id}`, requestOptions)
            .then(response => response.json())
            .then(data => onTestDataChange());
    }

    return (
        <Container>
            {details}
        </Container>
    );
}

export default TestPreviewList;