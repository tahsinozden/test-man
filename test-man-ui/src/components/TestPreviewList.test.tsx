import renderer from "react-test-renderer";
import TestPreviewList from "./TestPreviewList";
import TestDetail from "../models/TestDetail";
import {render, screen} from "@testing-library/react";
import "../jestGlobalMocks"

describe("TestPreviewList component", () => {
    it('renders correctly when there are no tests', () => {
        const tree = renderer
            .create(<TestPreviewList onTestDataChange={() => {
            }} testDetails={[]}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly when there are some tests', () => {
        const testDetails = [
            new TestDetail(1, "name1", "Passed", ""),
            new TestDetail(2, "name2", "Failed", ""),
        ];
        const tree = renderer
            .create(<TestPreviewList onTestDataChange={() => {
            }} testDetails={testDetails}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('shows test list correctly when there are some tests', () => {
        // given
        const testDetails = [
            new TestDetail(1, "name1", "Passed", ""),
            new TestDetail(2, "name2", "Failed", ""),
        ];

        // when
        render(<TestPreviewList onTestDataChange={() => {
        }} testDetails={testDetails}/>);

        // then
        let listItem = screen.getByText("name1");
        expect(listItem).toBeInTheDocument();

        listItem = screen.getByText("name2");
        expect(listItem).toBeInTheDocument();
    });
})
