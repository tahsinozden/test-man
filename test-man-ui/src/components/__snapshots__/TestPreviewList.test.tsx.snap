// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestPreviewList component renders correctly when there are no tests 1`] = `
<section
  className="ant-layout"
/>
`;

exports[`TestPreviewList component renders correctly when there are some tests 1`] = `
<section
  className="ant-layout"
/>
`;
